CircleciScripts/generate_documentation.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+# Creates Swift and Objective-C documentation by module, packages all modules
+# into a ZIP file, and leaves the result in the ${DOCS_TMP} directory, which is
+# safe to leave while checking out the gh-pages directory
+
+set -x
+
+SDK_VERSION="2.12.3"
+
+if [[ -z $SOURCE_ROOT ]] ; then
+  SOURCE_ROOT=$( pwd )
+fi
+
+DOCS_TMP=${SOURCE_ROOT}/docs_tmp
+
+# remove everything generated by this script
+function cleanup {
+  rm -rf ${DOCS_TMP}
+}
+
+if [ -n $1 ] && [ "$1" == "clean" ] ; then
+	cleanup
+	exit 0
+fi
+
+cleanup
+
+cd "$SOURCE_ROOT"
+
+mkdir -p ${DOCS_TMP}
+
+# Create destination (see pre_start_simulator.sh)
+test_device_id=$( bash ${SOURCE_ROOT}/CircleciScripts/get_circleci_test_device_id.sh )
```

Comment:
I always used backticks as well, but recently have been favoring the `$( )` construction because it's easier to visually distinguish from backticks. But I'll add the return val check